src/DecryptionFailureTracker.ts

```diff
@@ -158,10 +158,16 @@ export class DecryptionFailureTracker {
      *
      * @param {function} errorCodeMapFn The function used to map decryption failure reason  codes to the
      * `trackedErrorCode`.
+     *
+     * @param {boolean} dontTrackReportedEvents Don't check if we have already reported
+     * an event. This is only used for tests, to avoid possible false positives from
+     * the Bloom filter. This should be set to `true` for all tests except for those
+     * that specifically test the `reportedEvents` functionality.
      */
     private constructor(
         private readonly fn: TrackingFn,
         private readonly errorCodeMapFn: ErrCodeMapFn,
+        private readonly dontCheckReportedEvents: boolean = false,
```

Comment:
The double-negative here is making my head explode. Can we call it `checkReportedEvents` and have it default to true?

Reply:
done